pkg/debuginfo/metadata.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2022 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package debuginfo
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"path"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/thanos-io/objstore"
+)
+
+var (
+	ErrDebugInfoMetadataShouldExist            = errors.New("debug info metadata should exist")
+	ErrDebugInfoMetadataExpectedStateUploading = errors.New("debug info metadata state should be uploading")
+)
+
+type metadataState int64
+
+const (
+	// There was an unexpected error. The error will be filled in the return value.
+	metadataStateError metadataState = iota
```

Comment:
The `0` value shouldn't be overloaded with one that has semantic meaning. I would suggest we use an `Unknown` state here, which we never explicitly set.